book/configuration.nim

```diff
@@ -0,0 +1,54 @@
+import nimib, nimibook
+import std / [os, strutils, strscans, strformat]
+
+nbInit
+nbUseNimibook
+
+# the following is very dependent on how source is written
+proc readBookConfigFields: seq[string] =
+  # current folder is docs
+  let src = "../src/nimibook/types.nim"
```

Comment:
Unless I misunderstood something, why does the book rely upon parsing a file in ``src`` ?

I think we should keep ``book`` and ``src`` separate as much as possible.

Reply:
Also relative path based on "../" will break with Nimble; but that may be something to fix later on

Reply:
I was also confused about the src/book thing until I realized the file wasn't some kind of configuration file butit rather just one of the pages in the book which did some parsing of the doc-comments in src. 🤦‍♂️

Reply:
This `book` being the documentation of what happens in `src`, I think it is fair to make `book` depend on `src` to avoid repeating stuff and keep documentation up to date. The moment you add a new field to `Book` object implementing another feature you will get the corresponding documentation. For sure we should avoid having `src` depend on `book` folder. `book` folder should be forgotten when installed by nimble so the relative paths there should not matter.

Regarding the implementation, it is indeed a bit hacky (main drawback being that how you add and document fields in `Book` object impacts the documentation), but I think it is also rather straightforward. Other options I had in mind for implementation of this autodocumenting configurations were:

1. `nim jsondoc ...` and parse json file (will also depend directly on src)
2. use `macros.getImpl` and parse AST (this depends only on import nimibook, but since we will not have nimibook as a dependency here, it will still ultimately depend on src)

The only option to not depend on `src` folder is to duplicate documentation in code and `book` and I would prefer to avoid this unless there are specific issues.

Reply:
Options 2 was what I had in mind to do this but I understand your reasoning. 

I think in the long run we need a macros so that we can extract information from the AST for using Nimibook as a API documentation generator (replacement of Nim doc).

But it's a rather complex feature so maybe I think we should merge this first and improve upon it later.